docs/sidechain/spec.md

```diff
@@ -0,0 +1,945 @@
+# Sidechains
+
+## Overview
+
+This document defines the ledger objects, transactions, RPC commands, and
+additional servers that are used to support sidechains on the XRP ledger. A
+sidechain is an independent ledger. It has its own validators and can have its
+own set of custom transactions. Importantly, there is a way to move assets from
+the mainchain to the sidechain and there is a way to return those assets from
+the sidechain to the mainchain. This key operation is called a cross-chain
+transfer. A cross-chain transfer is not a single transaction. It happens on two
+chains, requires multiple transactions and involves an additional server type
+called a "witness". This key operation is described in a section below.
+
+After the mechanics of a cross-chain transfer are understood, an overview of
+some supporting transactions are described. This includes transactions to create
+sidechains, add signatures, create accounts across chains, and change sidechain
+parameters.
+
+The next sections describe the witness server, how to set-up a sidechain, error
+handling, and trust assumptions.
+
+Finally, a detailed reference for the new ledger objects and transactions are
+presented as a reference.
+
+## Nomenclature
+
+Cross-chain transfer: A transaction that moves assets from the mainchain to the
+sidechain, or returns those assets from the sidechain to the mainchain
+
+Cross-chain sequence number: A ledger object used to prove ownership of the
+funds moved in a cross-chain transfer.
+
+Door accounts: The account on the mainchain that is used to put assets into
+trust, or the account on the sidechain used to issue wrapped assets. The name
+comes from the idea that a door is used to move from one room to another and a
+door account is used to move from one chain to another.
+
+Mainchain: Where the assets originate and are put into trust.
+
+Sidechain: Where the assets from the mainchain are wrapped.
+
+Witness server: A server that listens for transactions on the main and side
```

Comment:
@seelabs  Can we use the term "attestor" instead of "witness"? I have suggested the same to @mvadari  and Amarantha too. I think attestor is better as the servers "attest" that a transaction took place.

Reply:
@mvadari  as we discussed in Slack, I have to run this with legal as well and make sure they don't see any risk whatever name we choose.